### ISSUE TITLE
Caret to wrong column after moving up/down after using search results window

### ISSUE THREAD

**Issue by sasumner** [notepad-plus-plus/notepad-plus-plus#8670]
### Description of the Issue
A bit hard to explain without some STR, so...just see the STR:

### Steps to Reproduce (STR) the Issue
1. Paste the following text into a new Notepad++ editing tab:
```
aaa bbb ccc dddd
aaa bbb ccc dddd
aaa bbb ccc dddd
aaa bbb ccc dddd
```
2. Observe your text:
![image](https://user-images.githubusercontent.com/30118311/89444328-8bf0e600-d71f-11ea-92c3-6183ea33d811.png)

3. Put your caret in Line 1 Column 1:
![image](https://user-images.githubusercontent.com/30118311/89445459-31f12000-d721-11ea-940f-d40de5b9887d.png)

4. Invoke the _Find_ dialog by pressing Ctrl+f.
5. Put `ccc` in _Find what_.
6. Run _Find All in Current Document_.
7. Observe _Find result_ window contents:
![image](https://user-images.githubusercontent.com/30118311/89444589-eb4ef600-d71f-11ea-8af3-acc74b7eabbc.png)
8. Double-click on the line with `Line 3` on it, in _Find results_.
9. Observe that focus moves back to the editor window, and the correct line and match is indicated:
![image](https://user-images.githubusercontent.com/30118311/89444769-35d07280-d720-11ea-972f-05248bcaae2b.png)
10. Press and release the up arrow key. Observe where caret moves to.

### Expected Behavior
I expected my caret to be placed here:
![image](https://user-images.githubusercontent.com/30118311/89444981-88aa2a00-d720-11ea-9e60-b7b35634dc13.png)

### Actual Behavior
My caret was placed here:
![image](https://user-images.githubusercontent.com/30118311/89445066-a9727f80-d720-11ea-9cd8-c293d2e8f9f1.png)

### Debug Information
Notepad++ v7.8.9   (32-bit)
Build time : Jul 15 2020 - 20:26:50
Path : C:\\............\npp.7.8.9.bin\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 1809
OS Build : 17763.1039
Current ANSI codepage : 1252
Plugins : mimeTools.dll NppConverter.dll NppExport.dll PythonScript.dll

**Comment by Uhf7** [comment]
I believe I  can reproduce it easier. Do STR steps 1, 2 and 3. Then Ctrl-F, but use simply "Find Next" _ccc_. The text cursor moves to the position after the _ccc_ in line 1. Then press the down arrow, and the cursor moves to the beginning of line 2 instead of to the end of _ccc_ in line 2.

**Comment by sasumner** [comment]
Perhaps this needs to be called after a match selection is set by code?:  https://www.scintilla.org/ScintillaDoc.html#SCI_CHOOSECARETX

**Comment by Uhf7** [comment]
Seems promising to call it after any find operation moved the cursor. The code in the Scintilla-Editor.cxx promises exactly this. And another thing speaks for it: If (in the _aaa_ _bbb_ _ccc_ example) the text cursor is after _aaa_ before the search, then it jumps to this column on the next up/down move.

**Comment by Uhf7** [comment]
Just quickly debugged Scintilla, `Editor::PositionUpOrDown`. The last parameter of it, `lastX`, makes the difference. And exactly this is what is set by `SCI_CHOOSECARETX`. So you only have to find the right places to call it. And, it has to be sure that the cursor (after find) is visible at the moment you call it, because it uses the current caret position and the current horizontal scroll position in its formula. See `Editor::SetLastXChosen()`. So first scroll to it, and then call `SCI_CHOOSECARETX`, I would guess.

**Comment by sasumner** [comment]
How could I be a Notepad++ user for 5+ years and never have noticed this annoying behavior before? :-)

**Comment by sasumner** [comment]
@Uhf7 

`Searching::displaySectionCentered()`

**Comment by sasumner** [comment]
Changes from this issue released in Notepad++ version 7.9
